Extension/src/LanguageServer/settings.ts

```diff
@@ -461,6 +463,33 @@ export class CppSettings extends Settings {
         return super.Section.get<boolean>("inlayHints.referenceOperator.showSpace") === true;
     }
 
+    public get otfDocsEnabled(): PromiseLike<boolean> {
+        // Check if the user has access to copilot.
+        return vscode.lm.selectChatModels({vendor: "copilot"}).then((models) => {
+            // Check if the setting is explicitly set to enabled or disabled.
+            const setting = super.Section.get<string>("onTheFlyDocsEnabled");
+
+            // If no models are returned, the user doesn't have access to copilot.
+            if (models.length === 0) {
+                // Register to update this setting if the user gains access to copilot.
+                vscode.lm.onDidChangeChatModels(() => {
+                    clients.ActiveClient.sendDidChangeSettings();
+                });
+                return false;
+            }
+
+            if (setting === "enabled") {
+                return true;
+            }
+            if (setting === "disabled") {
+                return false;
+            }
+
+            // Check for the feature flag.
+            return isFlightEnabled("cpp.otfDocs");
```

Comment:
I think it's okay to have a single flight given that this is a Boolean option, but can you double check with Sinem? I know in the experimentation platform there will actually be two flights (a treatment and a control).

Reply:
I believe we only need the one flag to check if it's enabled since we're defaulting to disabled, but I'll double check.